Reconstruction/RecCalorimeter/src/components/MergeCaloHitsTool.cpp

```diff
@@ -0,0 +1,106 @@
+#include "MergeCaloHitsTool.h"
+
+// our EDM
+#include "datamodel/CaloHit.h"
+
+DECLARE_TOOL_FACTORY(MergeCaloHitsTool)
+
+MergeCaloHitsTool::MergeCaloHitsTool(const std::string& type,const std::string& name, const IInterface* parent) 
+  : GaudiTool(type, name, parent)
+{
+  declareInterface<IMergeCaloHitsTool>(this);
+
+}
+
+MergeCaloHitsTool::~MergeCaloHitsTool()
+{
+}
+
+StatusCode MergeCaloHitsTool::initialize() {
+  StatusCode sc = GaudiTool::initialize();
+  if (sc.isFailure()) return sc;
+  return sc;
+
+}
+
+std::vector<fcc::CaloHit*> MergeCaloHitsTool::MergeOneCollection(const fcc::CaloHitCollection& aHits) {
+
+  //Vector of merged hits
+  std::vector<fcc::CaloHit*>  mergedHits;
```

Comment:
To avoid copies, could make sense to reserve here (either by using `mergedHits.reserve(aHits.size())` or to avoid over reserving by passing the size of the `m_edmCellsVector` from the algorithm).